sdk/servicebus/service-bus/src/track2/constructorHelpers.ts

```diff
@@ -0,0 +1,152 @@
+      const baseEntityPath = entityPathMatch![1]!;
+
+      if (hasSubscriptionName(auth)) {
+        // topic (from connection string) + sub
+        return {
+          context: createConnectionContextForConnectionString(auth.topicConnectionString, options),
+          entityPath: `${baseEntityPath}/Subscriptions/${auth.subscriptionName as string}`
+        };
+      } else {
+        throw new TypeError("Missing subscription name, required as part of connecting to a topic");
+      }
+    } else {
+      throw new TypeError("No entity name present in the connection string");
+    }
+  } else if (hasConnectionString(auth)) {
+    let entityPath: string;
+
+    if (hasQueueName(auth)) {
+      entityPath = auth.queueName;
+    } else if (hasTopicName(auth) && hasSubscriptionName(auth)) {
+      entityPath = `${auth.topicName}/Subscriptions/${auth.subscriptionName}`;
+    } else {
+      throw new TypeError("Missing fields when using TokenCredential authentication");
+    }
+
+    return {
+      context: createConnectionContextForConnectionString(auth.connectionString, options),
+      entityPath: entityPath
+    };
+  } else {
+    throw new TypeError("Unhandled set of parameters");
+  }
+}
+
+/**
+ * Temporary bit of conversion code until we can eliminate external usage of this
+ * enum.
+ * @param receiveMode
+ * @internal
+ * @ignore
+ */
+export function convertToInternalReceiveMode(
+  receiveMode: "peekLock" | "receiveAndDelete"
+): ReceiveMode {
+  switch (receiveMode) {
+    case "peekLock":
+      return ReceiveMode.peekLock;
+    case "receiveAndDelete":
+      return ReceiveMode.receiveAndDelete;
+  }
+}
+
+function hasHost(auth: any): auth is { host: string } {
+  return auth.host && typeof auth.host === "string";
+}
+
+function hasTokenCredentialAndHost(
+  auth: any
+): auth is { tokenCredential: TokenCredential; host: string } {
+  return isTokenCredential(auth.tokenCredential) && hasHost(auth);
+}
+
+function hasSubscriptionName(auth: any): auth is { subscriptionName: string } {
+  return auth.subscriptionName && typeof auth.subscriptionName === "string";
+}
+
+function hasQueueName(auth: any): auth is { queueName: string } {
+  return auth.queueName && typeof auth.queueName === "string";
+}
+
+function hasTopicName(auth: any): auth is { topicName: string } {
+  return auth.topicName && typeof auth.topicName === "string";
+}
+
+function hasQueueConnectionString(auth: any): auth is { queueConnectionString: string } {
+  return auth.queueConnectionString != null && typeof auth.queueConnectionString === "string";
+}
+
+function hasConnectionString(auth: any): auth is { connectionString: string } {
+  return auth.connectionString && typeof auth.connectionString === "string";
+}
+
+function hasTopicConnectionString(
+  auth: any
+): auth is {
+  topicConnectionString: string;
+} {
+  return auth.topicConnectionString != null && typeof auth.topicConnectionString === "string";
```

Comment:
Are you trying to ensure it's a non-empty string
```suggestion
  return auth.topicConnectionString && typeof auth.topicConnectionString === "string";
```
or just that it's a string
```suggestion
  return typeof auth.topicConnectionString === "string";
```

Same question applies to the other `has` functions above.

Reply:
No reason to not have the empty string check for free. Changed.